index.js

```diff
@@ -8,127 +8,192 @@ module.exports = {
+        'no-unused-vars': ['error', {
+            argsIgnorePattern: '^next$',
         }],
 
         // negated conditions are more difficult to understand. Code can be
         // made more readable by inverting the condition instead.
-        'no-negated-condition': [ 2 ],
-
-        'no-multi-spaces': ["error", {
-            "ignoreEOLComments": true,
+        'no-negated-condition': 'error',
 
+        'no-multi-spaces': ['error', {
+            ignoreEOLComments: true,
         }],
 
-        'max-len': [ 2, 100, 4, {
-            ignoreUrls: true
+        'max-len': ['error', {
+            code: 100,
+            tabWidth: 4,
+            ignoreUrls: true,
         }],
 
         // disallow weak equal signs (= instead of ==)
-        'eqeqeq': ["error", "always", { "null": "ignore" }],
+        'eqeqeq': ['error', 'always', { null: 'ignore' }],
 
         // jsdoc is required but doesn't have to be valid
         // this is in order to enforce some comments on classes and method
         // without requiring a lot of redundant boilerplate
-        'require-jsdoc': [ 2 ],
-        'valid-jsdoc': [ 0 ],
+        'require-jsdoc': 'error',
+        'valid-jsdoc': 'off',
 
         // semi-colons not required (standardjs.com)
-        'semi': [ 0 ],
+        'semi': ['error', 'never'],
 
         // trailing commas are required
-        'comma-dangle': ["error", "always-multiline"],
+        'comma-dangle': ['error', {
+            arrays: 'always-multiline',
+            objects: 'always-multiline',
+            imports: 'always-multiline',
+            exports: 'always-multiline',
+            functions: 'always-multiline',
+        }],
 
         // allow spacing around semi-colons
         // used for spacing for-loops for readability
-        'semi-spacing': [ 0 ],
+        'semi-spacing': 'off',
 
         // allow multiple empty lines for spacious code
-        'no-multiple-empty-lines': [ 0 ],
+        'no-multiple-empty-lines': ['error', { max: 2 }],
 
         // allow inline comments
-        'no-inline-comments': [ 0 ],
+        'no-inline-comments': 'off',
 
         // allow padded blocks where it makes sense
-        'padded-blocks': [ 0 ],
+        'padded-blocks': 'off',
 
         // allow TODO and FIXME comments
-        'no-warning-comments' : [ 0 ],
+        'no-warning-comments': 'off',
 
         // allow usage of the `var` keyword
-        'no-var': [ 0 ],
+        'no-var': 'off',
 
         // don't require parentheses for single argument arrow functions
-        'arrow-parens': [ 0 ],
+        'arrow-parens': 'off',
 
         // allow usage of `arguments` instead of only rest params (...args)
-        'prefer-rest-params': [ 0 ],
+        'prefer-rest-params': 'off',
 
         // allow the usage of `apply()` instead of only spread operator
-        'prefer-spread': [ 0 ],
+        'prefer-spread': 'off',
 
         // allow usage of `this` keyword outside of classes/class-like objects.
```

Comment:
Why are we allowing it  ?

Reply:
I'm keeping all these exceptions as for now because it would be overwhelming to revisit them all at once.